presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -187,6 +187,14 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
             throw new PrestoException(GENERIC_INTERNAL_ERROR, "SchedulingPolicy is bucketed, but BucketHandle is not present");
         }
 
+        if (bucketHandle.isPresent()) {
+            if (bucketHandle.get().getReadBucketCount() > bucketHandle.get().getTableBucketCount()) {
+                throw new PrestoException(
```

Comment:
I would throw UNSUPPORTED:

```
                throw new PrestoException(	
                        NOT_SUPPORTED,	
                         format("readBucketCount (%s) is greater than the tableBucketCount (%s). This is not supported yet.",  bucketHandle.get().getReadBucketCount(), bucketHandle.get().getTableBucketCount()));
            }```

Reply:
Generally this would be thrown because of an internal engine error - rather than incorrect input. Hence, I am throwing `GENERIC_INTERNAL_ERROR`. `UNSUPPORTED` indicates that this query shape is not accepted, where as we just need to use the correct bucket join mechanism.